server/mongo_data_access.go

```diff
@@ -97,18 +152,54 @@ func (dal *mongoDataAccessLayer) Delete(id, resourceType string) error {
 	}
 
 	collection := dal.Database.C(models.PluralizeLowerResourceName(resourceType))
-	return convertMongoErr(collection.RemoveId(bsonID.Hex()))
+	err = collection.RemoveId(bsonID.Hex())
+
+	if err == nil && dal.hasInterceptorsForVerbAndType("DELETE", resourceType) {
+		// Although this is a delete operation we need to get the resource first so we can
+		// run any interceptors on the resource before it's deleted.
+		resource, err := dal.Get(id, resourceType)
```

Comment:
Ummm... how does this work?  At this point you already deleted the resource, so... I don't think you can get it now.  You'll probably have to check if there is an interceptor (and get the data) _before_ you delete it.